compiler/rustc_codegen_llvm/src/gotoc/rvalue.rs

```diff
@@ -656,16 +657,32 @@ impl<'tcx> GotocCtx<'tcx> {
                 let src_goto_expr = self.codegen_operand(o);
                 let src_mir_type = self.operand_ty(o);
                 let dst_mir_type = t;
-                let dst_goto_expr = self.cast_sized_expr_to_unsized_expr(
-                    src_goto_expr.clone(),
-                    src_mir_type,
-                    dst_mir_type,
-                );
-                dst_goto_expr.unwrap_or(src_goto_expr)
+                self.cast_to_unsized_expr(src_goto_expr.clone(), src_mir_type, dst_mir_type)
+                    .unwrap_or(src_goto_expr)
             }
         }
     }
 
+    fn cast_to_unsized_expr(
+        &mut self,
+        src_goto_expr: Expr,
+        src_mir_type: Ty<'tcx>,
+        dst_mir_type: Ty<'tcx>,
+    ) -> Option<Expr> {
+        // Check if the cast is from a vtable fat pointer to another
```

Comment:
Can this ever happen between two different slice type unsized? Maybe we should check for that.